job_templates/ci_job.xml.em

```diff
@@ -192,7 +192,11 @@ echo "# BEGIN SECTION: Build Dockerfile"
 @[if os_name == 'linux-aarch64']@
 sed -i 's+^FROM.*$+FROM aarch64/ubuntu:xenial+' linux_docker_resources/Dockerfile
 sed -i 's+apt-get update+(apt-get update || true)+' linux_docker_resources/Dockerfile
+@[  if turtlebot_demo]@
+docker build --build-arg PLATFORM=arm --build-arg INSTALL_TURTLEBOT2_DEMO_DEPS=true -t ros2_batch_ci_turtlebot_demo linux_docker_resources
```

Comment:
If this gets any more options it might be worth breaking out a function that generates the correct `docker build` invocation for each combo. I don't think it needs to be refactored yet though.